src/NamedDims.jl

```diff
@@ -1,5 +1,11 @@
 module NamedDims
+using Base: @propagate_inbounds
+using Statistics
 
-greet() = print("Hello World!")
+export NamedDimsArray, name2dim, dim_names
```

Comment:
Seems weird to only partially follow the `_` convention.

Reply:
i prefer `a2b` naming only for `Dict`s but I don't have a good alternative and it's only an internal function

Reply:
so `name_to_dim`?
I am ok this that.

Reply:
`get_dim`?
`dim`?

Reply:
i quite to like `dim(names)` and `dim(names, name)` ...but maybe it's scarily short?

Reply:
We have `axes` it is only slightly shorter.

Reply:
Can't do `dims`,
it is a kwarg on functions that want  to call this.

e.g.  
`sum(xs::NamedDimArray; dims)=sum(parent(xs), dims(xs, dims))`
does  not work; and
`sum(xs::NamedDimArray; dims)=sum(parent(xs), NamedDimArray.dims(xs, dims))`
is ugly.